tensorzero-core/tests/e2e/datasets/tool_params.rs

```diff
@@ -331,8 +335,8 @@ async fn test_datapoint_update_tool_params() {
     // Verify updated tool_params (flattened at top level)
     assert_eq!(
         dp["allowed_tools"],
-        json!(["get_temperature", "updated_tool"]),
-        "allowed_tools should be updated"
+        json!(["get_temperature"]),
+        // Note: we no longer update updated tools because additional_tools are no longer added by default
```

Comment:
nit: I think we can just take this comment out since it references historical behavior that doesn't apply anymore.